Resources/Computing/BatchSystems/SLURM.py

```diff
@@ -59,8 +60,16 @@ def submitJob(self, **kwargs):
       # By default, all the environment variables of the submitter node are propagated to the workers
       # It can create conflicts during the installation of the pilots
       # --export restricts the propagation to the PATH variable to get a clean environment in the workers
-      cmd += "sbatch --export=PATH -o %s/%%j.out --partition=%s -n %s %s %s " % (
-          outputDir, queue, numberOfProcessors, submitOptions, executable)
+      cmd += "sbatch --export=PATH "
+      cmd += "-o %s/%%j.out " % outputDir
+      cmd += "-e %s/%%j.err " % errorDir
+      cmd += "--partition=%s " % queue
+      # One pilot (task) per node, allocating a certain number of cpus
+      cmd += "--ntasks-per-node=1 "
+      cmd += "--nodes=%s " % numberOfNodes
+      cmd += "--cpus-per-task=%s " % numberOfProcessors
+      # Additional options
+      cmd += "%s %s" % (submitOptions, executable)
```

Comment:
Dare we change this to use `subprocess` instead of `commands` (it's been deprecated since Python 2.6).

Reply:
I used the `DIRAC.Subprocess` class that calls the `subprocess` library. I noticed that both `DIRAC.Subprocess` and `subprocess` are used through DIRAC. Which one should be used?

Reply:
`DIRAC.Subprocess` is a wrapper around `subprocess` standard library. The 2 are basically interchangeable, but at least if you are using `DIRAC.Subprocess` you are sure that `subprocess32` is used (which is a backport of the `subprocess` module from python 3.

Reply:
Alright thanks.